posthog/temporal/data_modeling/workflows/execute_dag.py

```diff
@@ -0,0 +1,337 @@
+                continue
+
+            # execute child workflows in parallel for this level
+            child_handles = []
+            for node_id in execute_nodes:
+                handle = await temporalio.workflow.start_child_workflow(
+                    MaterializeViewWorkflow.run,
+                    MaterializeViewWorkflowInputs(
+                        team_id=inputs.team_id,
+                        dag_id=inputs.dag_id,
+                        node_id=node_id,
+                    ),
+                    id=f"materialize-{inputs.dag_id}-{node_id}-{temporalio.workflow.now().isoformat()}",
+                    retry_policy=temporalio.common.RetryPolicy(
+                        maximum_attempts=1,  # retries handled within child workflow
+                    ),
+                )
+                child_handles.append((node_id, handle))
+
+            # wait for all child workflows in this level to complete
+            for node_id, handle in child_handles:
+                try:
+                    result: MaterializeViewWorkflowResult = await handle
+                    node_results.append(
+                        NodeResult(
+                            node_id=node_id,
+                            success=True,
+                            rows_materialized=result.rows_materialized,
+                            duration_seconds=result.duration_seconds,
+                        )
+                    )
+                    temporalio.workflow.logger.info(
+                        f"Node {node_id} materialized successfully",
+                        extra={"rows_materialized": result.rows_materialized, **inputs.properties_to_log},
+                    )
+                except temporalio.exceptions.ChildWorkflowError as e:
+                    failed_node_set.add(node_id)
+                    error_message = str(e.cause) if e.cause else str(e)
+                    node_results.append(
+                        NodeResult(
+                            node_id=node_id,
+                            success=False,
+                            error=error_message,
+                        )
+                    )
+                    temporalio.workflow.logger.error(
+                        f"Node {node_id} failed to materialize: {error_message}",
+                        extra=inputs.properties_to_log,
+                    )
+                except Exception as e:
+                    capture_exception(e)
+                    failed_node_set.add(node_id)
+                    node_results.append(
+                        NodeResult(
+                            node_id=node_id,
+                            success=False,
+                            error=str(e),
+                        )
+                    )
+                    temporalio.workflow.logger.error(
+                        f"Node {node_id} failed with unexpected error: {str(e)}",
+                        extra=inputs.properties_to_log,
+                    )
+
```

Comment:
I think all the above logic should probably be in an activity - can we start child workflows from within an activity?